backend/app/project_handler_test.go

```diff
@@ -0,0 +1,185 @@
+package app
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/gin-gonic/gin"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCreateProject(t *testing.T) {
+	dbFilePath := "./test.db"
+	app, err := NewApp(dbFilePath)
+	assert.Nil(t, err)
+
+	app.router = gin.Default()
+	app.setRoutes()
+
+	t.Run("valid", func(t *testing.T) {
+		input := createProjectInput{Name: "new project"}
+		payload, err := json.Marshal(input)
+		assert.Nil(t, err)
+
+		req, err := http.NewRequest("POST", "/project", bytes.NewBuffer(payload))
+		assert.Nil(t, err)
+		req.Header.Set("Content-Type", "application/json")
+
+		w := httptest.NewRecorder()
+
+		app.router.ServeHTTP(w, req)
+
+		assert.Equal(t, http.StatusCreated, w.Code)
+	})
+
+	t.Run("invalid", func(t *testing.T) {
+		input := createProjectInput{Name: ""}
+		payload, err := json.Marshal(input)
+		assert.Nil(t, err)
+
+		req, err := http.NewRequest("POST", "/project", bytes.NewBuffer(payload))
+		assert.Nil(t, err)
+		req.Header.Set("Content-Type", "application/json")
+
+		w := httptest.NewRecorder()
+
+		app.router.ServeHTTP(w, req)
+
+		assert.Equal(t, http.StatusBadRequest, w.Code)
+	})
+}
+
+func TestUpdateProject(t *testing.T) {
+	dbFilePath := "./test.db"
+	app, err := NewApp(dbFilePath)
+	assert.Nil(t, err)
+
+	app.router = gin.Default()
+	app.setRoutes()
+
+	t.Run("valid", func(t *testing.T) {
+		input := createProjectInput{Name: "updated project"}
+		payload, err := json.Marshal(input)
+		assert.Nil(t, err)
+
+		req, err := http.NewRequest("PUT", "/project/1", bytes.NewBuffer(payload))
```

Comment:
tests should not depend on each other this way, each tests should be able to run on its own. state should be isolated between tests and tests should not leave artifacts behind them.